openaerostruct/docs/advanced_features/scripts/basic_2_sec_AS.py

```diff
@@ -0,0 +1,213 @@
+
+# Connect the B-spline component to the section B-splines
+connect_multi_spline(prob, section_surfaces, sec_twist_cp, "twist_cp", "twist_bspline", surface["name"])
+
+# docs checkpoint 3
+
+# Create and add a group that handles the geometry for the
+# aerostructual multi-section lifting surface
+multi_geom_group = MultiSecAerostructGeometry(surface=surface)
+prob.model.add_subsystem(surface["name"], multi_geom_group)
+
+name = surface["name"]
+
+point_name = "AS_point_0"
+
+# Create the aero point group and add it to the model
+AS_point = AerostructPoint(surfaces=[surface])
+
+prob.model.add_subsystem(
+    point_name,
+    AS_point,
+    promotes_inputs=[
+        "v",
+        "alpha",
+        "Mach_number",
+        "re",
+        "rho",
+        "CT",
+        "R",
+        "W0",
+        "speed_of_sound",
+        "empty_cg",
+        "load_factor",
+    ],
+)
+
+com_name = point_name + "." + name + "_perf"
+prob.model.connect(name + ".local_stiff_transformed", point_name + ".coupled." + name + ".local_stiff_transformed")
+prob.model.connect(name + ".nodes", point_name + ".coupled." + name + ".nodes")
+
+# Connect aerodyamic mesh to coupled group mesh
+prob.model.connect(name + ".mesh", point_name + ".coupled." + name + ".mesh")
+
+# Connect performance calculation variables
+prob.model.connect(name + ".radius", com_name + ".radius")
+prob.model.connect(name + ".thickness", com_name + ".thickness")
+prob.model.connect(name + ".nodes", com_name + ".nodes")
+prob.model.connect(name + ".cg_location", point_name + "." + "total_perf." + name + "_cg_location")
+prob.model.connect(name + ".structural_mass", point_name + "." + "total_perf." + name + "_structural_mass")
+prob.model.connect(name + ".t_over_c", com_name + ".t_over_c")
+
+
+# docs checkpoint 4
+
+prob.driver = om.ScipyOptimizeDriver()
+prob.driver.options["optimizer"] = "SLSQP"
+prob.driver.options["tol"] = 1e-6
+prob.driver.options["disp"] = True
+prob.driver.options["maxiter"] = 1000
+
+
+# Setup problem and add design variables, constraint, and objective
+prob.model.add_design_var("twist_bspline.twist_cp_spline", lower=-10.0, upper=15.0)
+prob.model.add_design_var("surface.thickness_cp", lower=0.01, upper=0.5, scaler=1e2)
+prob.model.add_constraint("AS_point_0.surface_perf.failure", upper=0.0)
+prob.model.add_constraint("AS_point_0.surface_perf.thickness_intersects", upper=0.0)
+
+# Add design variables, constraints, and objective on the problem
+prob.model.add_design_var("alpha", lower=-10.0, upper=10.0)
+prob.model.add_constraint("AS_point_0.L_equals_W", equals=0.0)
+prob.model.add_objective("AS_point_0.fuelburn", scaler=1e-5)
+
+
+# Set up the problem
+prob.setup(check=True)
+
+# Run the optimization
+optResult = prob.run_driver()
+# om.n2(prob, show_browser=False)
+
+
+# docs checkpoint 5
+
+# Get the unified mesh and plot the results
+meshUni = prob.get_val(name + "." + "surface_unification" + "." + name + "_uni_mesh")
+
+
+def plot_meshes(meshes):
```

Comment:
You don't need function here as you only use this once in this script. Alternatively you can import this from another file